src/Migrator.php

```diff
@@ -184,4 +199,22 @@ private function createBuilder(?MigrationInformerInterface $informer = null): Mi
             $informer ?? $this->informer,
         );
     }
+
+    private function transaction(Closure $operation): void
+    {
+        $transaction = $this->db->beginTransaction();
+
+        try {
+            $operation();
+            $transaction->commit();
+        } catch (Exception $e) {
+            if (
+                $this->db->getDriverName() === 'mysql' &&
+                $e->getMessage() !== 'Failed to commit transaction: transaction was inactive.'
```

Comment:
In this case, the migration will not be applied, but there will also be no exception? If yes - it's problem, if migration not applied, then migration process must be break.

Reply:
In this case the migration has not failed, but `MySQ`L makes the commit automatically, the error is because the commit was already done.

Reply:
Fixed.